waspc/src/Wasp/Package.hs

```diff
@@ -0,0 +1,91 @@
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Wasp.Package
+  ( Package (..),
+    getPackageProc,
+  )
+where
+
+import Control.Monad.Extra (unlessM)
+import StrongPath (Abs, Dir, File, Path', Rel, fromAbsDir, fromAbsFile, reldir, relfile, (</>))
+import System.Directory (doesDirectoryExist)
+import System.Exit (ExitCode (ExitFailure, ExitSuccess), exitFailure)
+import System.IO (hPutStrLn, stderr)
+import qualified System.Process as P
+import Wasp.Data (DataDir)
+import qualified Wasp.Data as Data
+import Wasp.Node.Version (getAndCheckNodeVersion)
+
+data Package
+  = DeployPackage
+  | TsInspectPackage
+
+data PackagesDir
+
+data PackageDir
+
+data PackageScript
+
+packagesDirInDataDir :: Path' (Rel DataDir) (Dir PackagesDir)
+packagesDirInDataDir = [reldir|packages|]
+
+packageDirInPackagesDir :: Package -> Path' (Rel PackagesDir) (Dir PackageDir)
+packageDirInPackagesDir DeployPackage = [reldir|deploy|]
+packageDirInPackagesDir TsInspectPackage = [reldir|ts-inspect|]
+
+scriptInPackageDir :: Path' (Rel PackageDir) (File PackageScript)
+scriptInPackageDir = [relfile|dist/index.js|]
+
+-- | Get a 'P.CreateProcess' for a particular package.
+--
+-- These packages are built during CI/locally via the @tools/install_packages_to_data_dir.sh@
+-- script.
+--
+-- If the package does not have its dependencies installed yet (i.e. after they
+-- just installed a Wasp version), we install the dependencies.
+getPackageProc :: Package -> [String] -> IO P.CreateProcess
```

Comment:
Ok all makes sense!
The thing is, naming is really bad on the `System.Process` side. `CreateProcess` -> pretty bad name, would be better if it was `ProcessCreator` or smth like that.
I forgot that that function is called `proc`, so seeing `getPackageProc` didn't help me much, I wasn't sure what `proc` their means. So I wouldn't stick to how they are calling it, instead would try to come up with a better name.

Ideas:
- `preparePackageForCalling` -> I like that it hides info about what it returns -> one can check that in the type. But, it also sounds like it does something every time, while it doesn't, it will do npm install and stuff only if needed, it doesn't do heavy prepartion every time.
- `ensurePackageIsPreparedForCalling` -> A bit better, `ensure` for me indicates it will ensure some conditions, which might need it will just confirm them, or actually produce them. But it sounds like it doesn't return anything. Which is not terrible, but still.
- `createPackageProcess` -> Plays on the idea that it returns `CreateProcess`. More haskellish, because if you give it two args, it inded becomes create process for package, but I think it also hides quite a bit so I don't like that a lot. `CReateProcess` name is giving us trouble here.
- `getPackageProcessCreator` -> Hides details about preparing the package, but makes it clear what it returns. Could be ok, caller doesn't really need to know how happens in the background. Maybe a better name would be `obtainPackageProcessCreator` -> `obtain` indicates it might need some heavy lifting, unlike `get` which usually means there is not much going on.

Looking at all this, maybe `preparePackageForCalling` or `obtainPackageProcessCreator` are the best in my opinion.

Reply:
I renamed it to `getPackageProcessOptions`. I think there isn't much of a difference between get and obtain here, so I just chose the shorter one. And "process options" seems, to me, like a much better name than `CreateProcess`, because it literally is the options/settings that `createProcess` uses when it spawns a process.